### PR TITLE
Basic attack timing

### PR DESCRIPTION
**Pull request by Eein**
add timing hits for one-hit attacks - will likely lock up for R character that has projectile attack.

### CODE REVIEW

File: memory/new_dialog_manager.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+from memory.core import mem_handle
+
+
+class NewDialogManager:
```
Comment (orkaboy) [inline memory/new_dialog_manager.py:4]: Possibly add a comment here as to what this class tracks exactly.

File: route/battle_test/battle_test.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        super().__init__(
+            name="BattleTest",
+            children=[
+                SeqDelay(name="Wait for select screen", timeout_in_s=3.0),
+                BattleTestSequence(),
+            ],
+        )
+
+
+class BattleTestSequence(SeqList):
```
Comment (orkaboy) [inline route/battle_test/battle_test.py:20]: Possibly add a comment here about which map the code is intended to be run on.

File: route/TAS.py [hunk 3]
```diff
@@ -154,3 +155,23 @@ def init_TAS(self):
         self.sequencer = SequencerEngine(
             window=self.window, config=self.config_data, root=TAS_root
         )
+
+
+class SoSBattleTestMenu(TASMenu):
```
Comment (orkaboy) [inline route/TAS.py:160]: I suggest overriding the `custom_gui()` method on this one and add some text in the GUI as to what location the function is supposed to be run.

File: route/evermist_island/intro.py [hunk 4]
```diff
@@ -34,7 +34,7 @@ def __init__(self):
                 SeqTurboMashUntilIdle(name="Wait for control"),
                 SeqLog(name="SYSTEM", text="We have control!"),
                 # TODO: Need to be able to do special ability (mash fails for Zale)
```
Comment (orkaboy) [inline route/evermist_island/intro.py:36]: I believe this is still the case, it will fail for Zale?

File: route/battle_test/battle_test.py (hunk 2 above)
Comment (Eein) [reply route/battle_test/battle_test.py:20]: Any! Its just to start the combat controller

File: route/evermist_island/intro.py (hunk 4 above)
Comment (Eein) [reply route/evermist_island/intro.py:36]: Until i add the timing ability for zale he has no way to do it. Its on a different skill index right now. Ill work on that next

File: route/battle_test/battle_test.py (hunk 2 above)
Comment (Eein) [reply route/battle_test/battle_test.py:20]: correct - i'll make some notes to make it clearer what its used for.

It's simply so i can start combat sequence whenever i want to test new changes; without messing with other sequences

**Pull request merged**